pkg/client/saramaclient_test.go

```diff
@@ -300,3 +300,200 @@ func TestSaramaClient_GetConsumerGroupsForTopic(t *testing.T) {
 
 	require.NoError(t, err)
 }
+
+func TestSaramaClient_GetEmptyTopicsSuccess(t *testing.T) {
+	admin := &MockClusterAdmin{}
+	mockClient := &MockSaramaClient{}
+	client := SaramaClient{admin: admin, client: mockClient}
+	brokerIDs := []int32{-1}
+	brokerMetaData := getLogDirsValue(sarama.ErrNoError)
+	brokers := []*sarama.Broker{sarama.NewBroker("broker-1:1234")}
+	mockClient.On("Brokers").Return(brokers).Once()
+	admin.On("DescribeLogDirs", brokerIDs).Return(brokerMetaData, nil).Once() //brokerid and response have different ids to check proper implementation
+
+	topics, err := client.GetEmptyTopics()
+
+	assert.NoError(t, err)
+	assert.ElementsMatch(t, []string{"topic-1"}, topics)
+	admin.AssertExpectations(t)
+}
+
+func TestSaramaClient_GetEmptyTopicsFailure(t *testing.T) {
+	admin := &MockClusterAdmin{}
+	mockClient := &MockSaramaClient{}
+	client := SaramaClient{admin: admin, client: mockClient}
+	brokerIDs := []int32{-1}
+	brokerMetaData := getLogDirsValue(sarama.ErrLeaderNotAvailable)
+	brokers := []*sarama.Broker{sarama.NewBroker("broker-1:1234")}
+	mockClient.On("Brokers").Return(brokers).Once()
+	admin.On("DescribeLogDirs", brokerIDs).Return(brokerMetaData, nil).Once() //brokerid and response have different ids to check proper implementation
+
+	topics, err := client.GetEmptyTopics()
+
+	assert.Error(t, err)
+	assert.True(t, errors.Is(err, sarama.ErrLeaderNotAvailable))
+	assert.Nil(t, topics)
+	admin.AssertExpectations(t)
+}
+
+func getLogDirsValue(errorCode sarama.KError) map[int32][]sarama.DescribeLogDirsResponseDirMetadata {
```

Comment:
this is too verbose/setup is part of this, so tests wouldn't be readable as what to expect. I would change this take some topic, string and return the struct, say

```
// partitionSize `partition_id:size,partition_id:size`
// `1:10,2:11,3:1`
buildlogDirsSizeMeta(errCode, topic, partitionSize string)

```